scripts/release/pick-patches.ts

```diff
@@ -0,0 +1,205 @@
+/* eslint-disable no-await-in-loop */
+import program from 'commander';
+import chalk from 'chalk';
+import { v4 as uuidv4 } from 'uuid';
+import type { GraphQlQueryResponseData } from '@octokit/graphql';
+import { graphql } from '@octokit/graphql';
+import ora from 'ora';
+
+import { simpleGit } from 'simple-git';
+
+program.name('pick-patches').description('Cherry pick patch PRs back to main');
+
+const logger = console;
+
+const OWNER = 'storybookjs';
+const REPO = 'monorepo-release-tooling-prototype';
+const SOURCE_BRANCH = 'next-v2';
+
+const graphqlWithAuth = graphql.defaults({
+  headers: { authorization: `token ${process.env.GH_TOKEN}` },
+});
+
+const git = simpleGit();
+
+interface PR {
+  number: number;
+  id: string;
+  branch: string;
+  title: string;
+  mergeCommit: string;
+}
+
+const LABEL = {
+  PATCH: 'patch',
+  PICKED: 'picked',
+  DOCUMENTATION: 'documentation',
+} as const;
+
+async function getUnpickedPRs(sourceBranch: string): Promise<Array<PR>> {
+  const result = await graphqlWithAuth<GraphQlQueryResponseData>(
+    `
+      query ($owner: String!, $repo: String!, $state: PullRequestState!, $order: IssueOrder!) {
+        repository(owner: $owner, name: $repo) {
+          pullRequests(states: [$state], labels: ["patch"], orderBy: $order, first: 50) {
+            nodes {
+              id
+              number
+              title
+              baseRefName
+              mergeCommit { 
+                abbreviatedOid
+              }
+              labels(first: 20) {
+                nodes {
+                  name
+                }
+              }
+            }
+          }
+        }
+      }
+    `,
+    {
+      owner: OWNER,
+      repo: REPO,
+      order: {
+        field: 'UPDATED_AT',
+        direction: 'DESC',
+      },
+      state: 'MERGED',
+    }
+  );
+
+  const {
+    pullRequests: { nodes },
+  } = result.repository;
+
+  const prs = nodes.map((node: any) => ({
+    number: node.number,
+    id: node.id,
+    branch: node.baseRefName,
+    title: node.title,
+    mergeCommit: node.mergeCommit.abbreviatedOid,
+    labels: node.labels.nodes.map((l: any) => l.name),
+  }));
+
+  const unpickedPRs = prs;
+  const branchPRs = unpickedPRs.filter((pr: any) => pr.branch === sourceBranch);
```

Comment:
```suggestion
  const branchPRs = prs.filter((pr: any) => pr.branch === sourceBranch);
```

Reply:
Why does `pr.branch === sourceBranch` even work? Shouldn't this instead check if the "picked" label is on the PR?

Reply:
or is it because you both want to filter on the base branch _and_ for unpicked PRs? I think we can use the `baseRefName` argument in the GraphQL query to do that.

Reply:
The reason it worked was because I had some uncommited local changes.

Reply:
Fixed now!